src/lib/integration_test_cloud_engine/coverage_manager.ml

```diff
@@ -0,0 +1,55 @@
+(* Submodule responsible for managing test coverage files. Responsible for code coverage raw files generatation from pods.
+   Requires tailored mina images with instrumentation and BISECT_SIGTEM env variable set.
+   Function traverse all applicable pods (those which has mina process) and kills them.
+   This operation dumps raw bissect files (.coverage) and downloads it to gcloud bucket.
+   WARNING: operations of test coverage retrieval is destructive (as it kills mina processes).
+*)
+open Integration_test_lib
+open Core
+
+module Coverage_manager = struct
+  type t = { logger : Logger.t }
+
+  let create ~logger = { logger }
+
+  let download_coverage_data_from_nodes t ~nodes =
+    let open Kubernetes_network in
+    let root = "/root/.mina-config" in
+    let open Malleable_error.Let_syntax in
+    [%log' info t.logger] "Generating test coverage data..." ;
+    let%bind coverage_files =
+      nodes |> Map.to_alist
+      |> Malleable_error.List.map ~f:(fun (_id, pod) ->
+             let pod_id = Node.id pod in
+             let%bind () = Node.stop pod in
+             [%log' info t.logger] "Mina process in '$pod' killed."
```

Comment:
Why is something called `Coverage_manager` killing pods?!

Reply:
I moved module to a function . There is also comment that gathering coverage is a destructive operation. Do you think i should rename more function name?